member_map.go

```diff
@@ -230,6 +234,42 @@ func (m *MemberMap) Alive(aliveMessage AliveMessage) (bool, error) {
 	return true, nil
 }
 
+// Handle Confirm Message
+
+func (m *MemberMap) Confirm(confirmMessage ConfirmMessage) (bool, error) {
+	m.lock.Lock()
+	defer m.lock.Unlock()
+
+	// Check whether Member ID in Confirm message is empty or not
+	if confirmMessage.ID == "" {
+		return false, ErrEmptyMemberID
+	}
+
+	// Get the existing member from the member list
+	mID := MemberID{confirmMessage.ID}
+	member, exist := m.members[mID]
+
+	// Check that the member exists
+	if !exist {
+		return false, nil
+	}
+
+	// {Dead Ml, inc=i} overrides
+	// - {Suspect Ml, inc=j}, any j
+	// - {Alive Ml, inc=j}, any j
+	status := member.Status
+	if status == Alive || status == Suspected {
+		member.Status = Dead
+		member.LastStatusChange = time.Now()
+		member.Incarnation = confirmMessage.Incarnation
```

Comment:
Now I think this logic should change into reducing suspect timeout or update member states into Suspect and start timer